backend/src/main/entity/User.ts

```diff
@@ -69,4 +70,12 @@ export class User {
     unique: true,
   })
   password: string;
+
+  hashPassword() {
+    this.password = bcrypt.hashSync(this.password, 8);
+  }
```

Comment:
What do you think about something like this?
```
  constructor(
    firstName: string,
    lastName: string,
    email: string,
    username: string,
    password: string
  ) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.username = username;
    this.password =bycrypt,hashSync(password, 8); // 
  }
```
This way we don't need to maintain the overhead of encrypting the password each time we create a new user, we can just assume once we create a user the password is encrypted.

Reply:
Then we also don't need the changes inside `Data.ts`

Reply:
Yeah, that's good idea! I want to ask you that if we use 'update' method of repository, is this constructor applied?

Reply:
Ah wait that's a good point, then we would have to manually do something like inside `update` method of repository: 
```
if (userUpdateDto.password) {
 userUpdateDto.password = bycript.hashSync(userUpdateDto.password, 8)
}
```

Reply:
Ok, I'll do that!